src/main/java/com/jpa/services/DeveloperService.java

```diff
@@ -0,0 +1,69 @@
+package com.jpa.services;
+
+import com.jpa.entity.Developer;
+import lombok.extern.log4j.Log4j;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Persistence;
+
+@Log4j
+public class DeveloperService implements Service<Developer> {
+
+    public Developer read(int id) {
+        EntityManager entityManager = Persistence.createEntityManagerFactory("JPA").createEntityManager();
+        Developer developer = new Developer();
+        try {
+            entityManager.getTransaction().begin();
+            developer = entityManager.find(Developer.class, id);
+            entityManager.getTransaction().commit();
+        } catch (Exception ex) {
+            entityManager.getTransaction().rollback();
```

Comment:
This code looks like a duplicate. You can find it in other methods. Could you move it into a seperate method and reuse?